content-review/impl/dao/src/main/java/org/sakaiproject/contentreview/dao/ContentReviewItemDao.java

```diff
@@ -56,7 +57,23 @@
 
 		return c.list();
 	}
-	
+
+	@SuppressWarnings("unchecked")
+	public List<Object[]> findByProviderGroupedBySite(Integer providerId) {
+
+		Criteria c = sessionFactory.getCurrentSession()
+				.createCriteria(ContentReviewItem.class)
+				.add(Restrictions.eq("providerId", providerId))
+				.setProjection( Projections.projectionList()
+							.add(Projections.distinct(Projections.property("siteId")))
+							.add(Projections.property("contentId"))
+						)
+				.addOrder(Order.desc("contentId"))
+				.setMaxResults(999);
+
+		return c.list();
```

Comment:
Returning a List<Object[]> is pretty vague.
This could be improved since this query knows that "siteId" and "contentId" are both Strings we could improve this api by returning a List<String[]> with:
`return c.list().stream().map(o -> Arrays.stream(o).map(Objects::toString).toArray(String[]::new)).collect(Collectors.toList());`

I would also update the methods javadoc to indicate what is in the String[].